Add `CupertinoTimerPicker` Interactive Example

TahaTesser opened the pull request:
This adds an official `CupertinoTimerPicker` example, this makes it much easier to try timer picker without navigating to large Flutter Gallery source code or relying on third-party websites.

Note:
I couldn't format the duration without importing `intl`. 

Related https://github.com/flutter/flutter/issues/72926



## Pre-launch Checklist

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I read and followed the [Flutter Style Guide], including [Features we expect every widget to implement].
- [x] I signed the [CLA].
- [x] I listed at least one issue that this PR fixes in the description above.
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] I added new tests to check the change I am making, or this PR is [test-exempt].
- [x] All existing and new tests are passing.

If you need help, consider asking for advice on the #hackers-new channel on [Discord].

<!-- Links -->
[Contributor Guide]: https://github.com/flutter/flutter/wiki/Tree-hygiene#overview
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[test-exempt]: https://github.com/flutter/flutter/wiki/Tree-hygiene#tests
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[Features we expect every widget to implement]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo#features-we-expect-every-widget-to-implement
[CLA]: https://cla.developers.google.com/
[flutter/tests]: https://github.com/flutter/tests
[breaking change policy]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes
[Discord]: https://github.com/flutter/flutter/wiki/Chat

TahaTesser committed:
Add CupertinoTimerPicker example

TahaTesser committed:
remove trailing space

TahaTesser committed:
Kick tests

gspencergoog reviewed (commented):
This looks good, but will need a test.

TahaTesser committed:
test and clean up

TahaTesser committed:
Update comment

TahaTesser committed:
Kick tests

TahaTesser commented:
@gspencergoog 
Added test and cleaned up 
Thank you so much for the review!

gspencergoog commented on examples/api/lib/cupertino/date_picker/cupertino_timer_picker.0.dart:
```diff
@@ -0,0 +1,122 @@
+                  const Text('Timer'),
+                  CupertinoButton(
+                    onPressed: () => _showDialog(
+                      CupertinoTimerPicker(
```
Some comments here about what you're trying to make with the `CupertinoTimePicker` might be useful.

Hixie commented:
@TahaTesser Is this PR still on your radar?

TahaTesser commented:
@Hixie 
Yes. I have been favoring material doc issues lately since I have been working with Material widgets.
Thanks for the ping! I will update this in a day.

TahaTesser committed:
Add comment and improve sample

TahaTesser commented:
@gspencergoog 
Added comments similar to my other example (CupertinoDatepicker) and changed `hms` to `hm` so to showcase non-default parameters in the example, update example title as well. Sorry for the delay, thanks for the review!

gspencergoog reviewed (approved):
![32384589-a60f0e74-c078-11e7-9bc1-e5b5287aea9d](https://user-images.githubusercontent.com/8867023/37737237-66e9c9dc-2d10-11e8-8797-13a405eace49.png)
